coralproject/talk-fe
# Connect backend

## What does this PR do?

Connects talk-fe to shelf, retrieving comments on page load and posting comments from the comment box.

## How do I test this PR?

1. Update to the latest version of shelf and talk
2. Start backend services using the [launch script](https://github.com/coralproject/talk/blob/master/install/local/launch.sh) in talk.
3. Post an item of type 'page' and id 'pageTest' to xenia:
`curl -X POST -d "{\"type\":\"page\", \"version\":1, \"data\":{\"comments\":[], \"url\":\"http://coralproject.net\"}, \"item_id\":\"pageTest\"}" http://localhost:16180/v1/item`
4. From talk-fe root directory: `npm install && lerna bootstrap`
5. `lerna-run npm test`, all tests should pass.
6. Navigate to `./packages/coral-embed-stream`
7. `npm dev-start` and open `http://localhost:3000` in a browser
8. An unstyled comment form should be visible
9. Posted comments should appear immediately
10. Reload the page, posted comments should be retrieved from the server
  - Comment order will not be supported until timestamps are added to the backend
